### ISSUE TITLE
Artifacts

### ISSUE THREAD

**Issue by Zarxrax** [dslisleedh/PLKSR#3]
Hi, I have been trying out PLKSR within [neosr ](https://github.com/muslll/neosr/)training framework. 
Sometimes the output will degrade into weird noise on parts of the image. 
Is this something that you encountered while working on PLKSR?
I wasn't sure if this was an issue with PLKSR itself, or perhaps a problem with its implementation in neosr. Or do you have any tips to prevent this kind of issue?
![ep02 015366_35000](https://github.com/dslisleedh/PLKSR/assets/16127268/eb10717c-b3b6-4fc2-be6b-903a3698dd3b)

**Comment by dslisleedh** [comment]
I appreciate your interest in PLKSR! I haven't encountered this problem because I only generated the resulting images after completing the training. I also checked the implementation by neosr, and it seems fine. One possible guess is that PLKSR's training can be quite hard because it doesn't use intermediate normalization to reduce latency. I wonder if adding a normalization layer or using a pre-trained model could address this issue.

**Comment by umzi2** [comment]
using pre-training instantly leads to artifacts, from the way I solved it, it’s a warm-up and a pre-trained discriminator
![изображение](https://github.com/dslisleedh/PLKSR/assets/131161226/dcfe91c8-e639-4026-9015-8b7acf09a644)

**Comment by umzi2** [comment]
just as it seemed to me that 2x models are less stable, and the more losses and the more unstable the discriminator and optimizer are, the more often these artifacts appear. I think plksr is just very sensitive and this is something to take into account when training. I really liked your architecture, but it’s really difficult to train and artifacts can appear in the most unexpected places and usually this happens on the same images

**Comment by dslisleedh** [comment]
I don't know about Real-image SR as I haven't trained it, but I agree that PLKSR is harder to train compared to other models. For example, compared to other models (e.g. DITN, SRFormer-light), PLKSR had to reduce the learning rate in the earlier iterations to avoid gradient explosion. I think this is probably a big weakness of PLKSR.
